### PR TITLE
[node]: Make stream destroy method signatures match

### PR DESCRIPTION
**Pull request by fluggo**
The `destroy` option when creating streams has the same signature as the _destroy method on the stream: a function accepting an optional error and a callback which also takes an optional error.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes:
  - https://medium.com/@CWMma/whats-new-for-streams-in-node-8-736d431083df
  - https://nodejs.org/api/stream.html#stream_implementing_a_writable_stream
  - https://nodejs.org/api/stream.html#stream_writable_destroy_err_callback
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by fluggo** [comment]
Also: I know it fails the tests, but that's due to unrelated errors in react-sortable-tree and redux-form.

File: types/node/index.d.ts [hunk 1]
```diff
@@ -5972,7 +5972,7 @@ declare module "stream" {
             encoding?: string;
             objectMode?: boolean;
             read?: (this: Readable, size?: number) => any;
-            destroy?: (error?: Error) => any;
+            destroy?: (error: Error | undefined, callback: (error?: Error) => void) => void;
```
Comment (Flarna) [inline types/node/index.d.ts]: Based on https://github.com/nodejs/node/blob/v10.7.0/lib/internal/streams/destroy.js#L32 I would expect `Error | null` here and related places.

Comment (fluggo) [reply types/node/index.d.ts]: Changing the definition to `null` causes errors in other packages expecting `undefined`:

* [amp](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/amp/amp-tests.ts#L31)
* [stream-chain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/stream-chain/stream-chain-tests.ts#L145)

Comment (fluggo) [reply types/node/index.d.ts]: I have pushed a fix to those tests as well.

**Review by uhop — APPROVED** [review]
The changes for stream-chain are correct.

**Pull request merged**